### DISCUSSION TITLE
Invalid memory address or nil pointer dereference

### DISCUSSION THREAD

**Discussion by shamilh2o** [h2oai/wave#1949] [Q&A] [answered]
Hey team, When I am using a wave app in HAMC I am getting this panic runtime error time to time. I have seen that, it is fixed in #1690 

> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x77ce50]

> goroutine 166 [running]:
> github.com/h2oai/wave.(*App).send(0xc000370000, 0x0, 0x0, 0xc00052c000, 0xaae820, 0x20, 0x20, 0x0, 0x0)
> 	/home/runner/work/wave/wave/app.go:87 +0x510
> github.com/h2oai/wave.(*App).forward(0xc000370000, 0x0, 0x0, 0xc00052c000, 0xaae820, 0x20, 0x20)
> 	/home/runner/work/wave/wave/app.go:66 +0x85
> github.com/h2oai/wave.(*Broker).resetClients.func1(0xc00052c000, 0xc000370000)
> 	/home/runner/work/wave/wave/broker.go:212 +0x65
> created by github.com/h2oai/wave.(*Broker).resetClients
> 	/home/runner/work/wave/wave/broker.go:211 +0xe5
> {"level":"error","cmd":"/resources/venv/waved","process":55,"error":"exit status 2","time":"2023-04-27T11:02:54Z","message":"Cmd terminated, killing all the other cmds"}
> {"level":"error","cmd":"/resources/venv/bin/uvicorn","process":64,"error":"signal: killed","time":"2023-04-27T11:02:54Z","message":"Cmd terminated, killing all the other cmds"}
> failed: failed running wave app: waved error: cmd error: exit status 2

From app end wave version is 0.24.2. The wave server defined in env I can see as 0.23.0. Is that can be the reason for this?
cc: @mturoci

**Comment by mturoci** [comment]
What makes you think it is related to #1690 ?

> From app end wave version is 0.24.2. The wave server defined in env I can see as 0.23.0. Is that can be the reason for this?

Wave app and wave server need to always be in sync to prevent unexpected behavior.

**Reply by shamilh2o** [reply]
Yes not exactly the same. I saw the description has some similar error lines regarding this `invalid memory address or nil pointer dereference`.

Anyway, do you know from where the wave server version is defining in envs? 
Do you think this occurs due to that mismatch or anything need to fix from Wave end?

**Reply by mturoci** [reply, marked as answer]
> Anyway, do you know from where the wave server version is defining in envs?

This is project dependent. We have no control over how people run Wave. You need to look into your project.

> Do you think this occurs due to that mismatch or anything need to fix from Wave end?

As I wrote above, the versions must be in sync. If you face a problem with having the versions synced, then it might be something we need to fix.

**Reply by shamilh2o** [reply]
Got it. Thanks @mturoci
